### ISSUE TITLE
Failure to connect to HA model

### ISSUE THREAD

**Issue by chris-sanders** [juju/python-libjuju#385]
Trying to use libjuju I've run across a specific controller which without fail causes libjuju to hang for a very long time and eventually crash out during a ping. I don't know that it would ever exit if the ping didn't fail at some point. I see early on that some web-sockets get closed, by the server I presume, but having checked the controller I can't find any issue with it. Juju cli works as expected and the site runs with out incident otherwise it's just libjuju that can't connect (or stay connected) to it.

Debug log with IP addresses and my credentials redacted: https://pastebin.canonical.com/p/j5J9wH3mgk/

I don't know a lot about websockets but looking up the error code 1009, which might be a red herring, I adjusted the max_frame_size on the model before connecting which makes no difference. The very same debug output is generated (and I put in a print statement to be sure it was getting adjusted, it is).

**Comment by SimonRichardson** [comment]
@chris-sanders are you passing in your own credentials or letting pylibjuju pick them up from `.local/share/juju`?

**Comment by chris-sanders** [comment]
It's getting picked up from the controller.yaml. There same yaml had 4 controllers defined and only this one fails to connect every time.
Two controllers always connect without issue, and one will connect to dome endpoints but fails in this same way on others. I managed to get that one too work by changing the first endpoint in the list to (these are HA)

**Comment by SimonRichardson** [comment]
> (these are HA)

This is the important piece, tbh I've yet to test pylibjuju in HA mode. I'll try and get some time to see if I spot the issue.

**Comment by ajkavanagh** [comment]
I've run into this just yesterday (libjuju 2.8.3).  Just spun when trying to do model.connect().  Connected fine to another model on the same controller.  This was not HA.  When looking at the logs I just saw it ping and ponging with the controller but not finishing the protocol (I'm guessing) as it never finished and returned the model object.

I also restarted the controller and db on the controller machine.  No effect.  I'm guessing something is different about the model in the DB and that was causing an issue?

I also tested the same controller/model against 2.8.2 and 2.8.1 with no success.  If I hit the issue again I'll try to collect some additional logging information.  Is there a way I can increase the logging at the controller?
